packages/@aws-cdk/aws-ec2/lib/user-data.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Instance User Data
+ */
+export abstract class UserData {
+  /**
+   * Create a userdata object for Linux hosts
+   */
+  public static forLinux(): UserData {
+    return new LinuxUserData();
+  }
+
+  /**
+   * Create a userdata object for Windows hosts
+   */
+  public static forWindows(): UserData {
+    return new WindowsUserData();
+  }
+
+  /**
+   * Add one or more commands to the user data
+   */
+  public abstract addCommands(...commands: string[]): void;
+
+  /**
+   * Render the UserData for use in a construct
+   */
+  public abstract render(): string;
+}
+
+class LinuxUserData extends UserData {
+  private readonly lines: string[] = [];
+
+  constructor() {
+    super();
+  }
+
+  public addCommands(...commands: string[]) {
+    this.lines.push(...commands);
+  }
+
+  public render(): string {
```

Comment:
The shebang line should be something that users can customize (e.g. `UserData.forLinux({ shebang: '#!/boom' })`).
https://github.com/awslabs/aws-cdk/issues/2674